article/test/model/LiveBlogCurrentPageTest.scala

```diff
@@ -132,6 +160,70 @@ class LiveBlogCurrentPageTest extends FlatSpec with Matchers {
     result.get.pinnedBlock should be(Some(expectedPinnedBlock))
   }
 
+  it should "return none when no summary exist in requestedBodyBlocks and the filter is on" in {
+    val blocks = fakeBlocks(5, 4)
+    val result = LiveBlogCurrentPage.firstPage(
+      2,
+      Blocks(
+        3,
+        Nil,
+        None,
+        Map(
+          CanonicalLiveBlog.firstPage -> blocks.take(3),
+          CanonicalLiveBlog.pinned -> List(),
+          CanonicalLiveBlog.timeline -> blocks.take(6),
+          CanonicalLiveBlog.oldestPage -> blocks.lastOption.toSeq,
+        ),
+      ),
+      true,
+    )
+
+    result should be(None)
+  }
+
+  it should "return none when no key events exist in requestedBodyBlocks and the filter is on" in {
+    val blocks = fakeBlocks(5, 0, 0, 4)
+    val result = LiveBlogCurrentPage.firstPage(
+      2,
+      Blocks(
+        3,
+        Nil,
+        None,
+        Map(
+          CanonicalLiveBlog.firstPage -> blocks.take(3),
+          CanonicalLiveBlog.pinned -> List(),
+          CanonicalLiveBlog.summary -> blocks.take(6),
+          CanonicalLiveBlog.oldestPage -> blocks.lastOption.toSeq,
+        ),
+      ),
+      true,
+    )
+
+    result should be(None)
+  }
+
+  it should "display a page with no blocks when key events and summary lists are empty and filter is on" in {
+    val blocks = fakeBlocks(5, 0, 0, 4)
+    val result = LiveBlogCurrentPage.firstPage(
+      2,
+      Blocks(
+        3,
+        Nil,
+        None,
+        Map(
+          CanonicalLiveBlog.firstPage -> blocks.take(3),
+          CanonicalLiveBlog.pinned -> List(),
+          CanonicalLiveBlog.timeline -> List(),
+          CanonicalLiveBlog.summary -> List(),
+          CanonicalLiveBlog.oldestPage -> blocks.lastOption.toSeq,
+        ),
+      ),
+      true,
+    )
+
+    should(result, currentPage = FirstPage(List(), true), pagination = None)
+  }
+
```

Comment:
This is great. Marjan and I discussed adding additional tests for the 2 other scenarios in the apply method to make this a fully comprehensive PR.

Reply:
I updated an existing test that was testing `findPageWithBlock` with filter switch on. So the test is now also having some summaries in it's input blocks list. 

I didn't get the chance to add test for `updates` function. But it may not be necessary since the `updates` is also using `applyFilters`  same as in`findPageWithBlock` which was covered in the above test. 

I'm happy to add the test on Thursday if you think would add benefit though